### PR TITLE
New field

### PR DESCRIPTION
**Pull request by jdiog0**
Added a new field for variable length of bytes in little-endian and hexadecimal (#1936). I'm using this in a new layer that I'm working on

### CODE REVIEW

File: scapy/fields.py [hunk 1]
```diff
@@ -1234,6 +1234,19 @@ def i2repr(self, pkt, x):
         return bytes_hex(x[:self.length_from(pkt)]).decode()
 
 
+class XLEStrLenField(XStrLenField):
+    def i2repr(self, pkt, x):
+        a = bytes_hex(x[:self.length_from(pkt)]).decode()
+        le = "".join(reversed([a[i:i + 2] for i in range(0, len(a), 2)]))
+        return hex(int("0x" + le, 16))
```
Comment (gpotter2) [inline scapy/fields.py]: Why reverse it again ?

Comment (jdiog0) [reply scapy/fields.py]: The idea is to have a long hexadecimal value that user can manipulate but automatically encode it as LE to send over the network. When the user wants to save a value b’\xaa\xbb’ it is stored as b’\xbb\xaa’ but when the packet is showed is presented b’\xaa\xbb’

Comment (gpotter2) [reply scapy/fields.py]: You should rather store it `b'\xaa\xbb`' and do the reverse part in `i2m` / `m2i`

File: scapy/fields.py [hunk 2]
```diff
@@ -1234,6 +1234,18 @@ def i2repr(self, pkt, x):
         return bytes_hex(x[:self.length_from(pkt)]).decode()
 
 
+class XLEStrLenField(XStrLenField):
+    def i2repr(self, pkt, x):
+        a = bytes_hex(x[:self.length_from(pkt)]).decode()
+        return hex(int("0x" + a, 16))
```
Comment (gpotter2) [inline scapy/fields.py]: It doesn't make any sense to have `0x` in front of a hex encoded bytes-string.

It does with numbers, but we have `lhex()` to handle it in that case.

**Comment by gpotter2** [comment]
Please fix the conflicts. Sorry for the delay

**Comment by enen92** [comment]
@gpotter2 is anything missing here? We'd like to use this (plus the enip layer) upstream in another project

**Pull request merged**